universal-application-tool-0.0.1/test/support/ProgramBuilder.java

```diff
@@ -202,13 +202,22 @@ public BlockBuilder withPredicate(PredicateDefinition predicate) {
       return this;
     }
 
+    /** Add a required question to the block. */
     public BlockBuilder withQuestion(Question question) {
       blockDefBuilder.addQuestion(
           ProgramQuestionDefinition.create(
               question.getQuestionDefinition(), Optional.of(programBuilder.programDefinitionId)));
       return this;
     }
 
+    public BlockBuilder withQuestion(Question question, boolean optional) {
```

Comment:
suggestion: might be clearer to have `withRequiredQuestion` and `withOptionalQuestion` methods - seems a little more readable

Reply:
I'll do a follow up PR for this